values.yaml

```diff
@@ -1,402 +1,144 @@
-# storageClassName -- Sets the storage class for all Coder services and user
-# environments. By default, the storageClassName is unspecified, and thus the
-# default StorageClass is used. If storageClassName is not specified and a
-# default StorageClass does not exist, then the deployment will fail. The
-# storageClass MUST support the ReadWriteOnce access mode.
-storageClassName: ""
-
-# serviceType -- See the following for the different serviceType options and
-# their use:
-# https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
-serviceType: "ClusterIP"
-
-# podSecurityPolicyName -- The name of the pod security policy to apply to all
-# Coder services and user environments. The optional ingress has its own field
-# for pod security policy as well.
-podSecurityPolicyName: ""
+# coderd -- Primary service responsible for all things Coder!
+coderd:
+  # coderd.image -- Injected by Coder during release.
+  image: ""
+  # coderd.replicas -- The number of Kubernetes Pod replicas.
+  replicas: 1
+  # coderd.replica -- Deploy a replica to geodistribute access to
+  # workspaces for lower latency.
+  replica:
+    # coderd.replica.enable -- Run coderd as a replica pointing to a primary deployment.
+    enable: false
+    # coderd.replica.primaryURL -- URL of the primary deployment.
+    primaryURL: ""
+  # coderd.securityContext -- Fields related to the container's security
+  # context (as opposed to the pod).
+  securityContext:
+    # coderd.securityContext.readonlyRootFilesystem -- Sets the root filesystem
+    # of the container to read-only. It is recommended to leave this setting enabled
+    # in production.
+    readOnlyRootFilesystem: true
+  # coderd.resources -- Kubernetes resource specification for coderd pods.
+  # To unset a value, set it to "". To unset all values, set resources to nil.
+  resources:
+    requests:
+      cpu: "250m"
+      memory: "512Mi"
+    limits:
+      cpu: "250m"
+      memory: "512Mi"
 
-# clusterDomainSuffix -- If you've set a custom default domain for your
-# cluster, you may need to remove or change this DNS suffix for service
-# resolution to work correctly.
-clusterDomainSuffix: ".svc.cluster.local"
+# envbox -- Required for running Docker inside containers. See requirements:
+# https://coder.com/docs/coder/v1.19/admin/workspace-management/cvms
+envbox:
+  # envbox.image -- Injected by Coder during release.
+  image: ""
 
-# contains configuration for the bundled ingress controller.
+# ingress -- Configure access to your deployment.
```

Comment:
I don't think we should use the ingress term here unless we are configuring an actual Ingress. For kube Service config, we should have a key like `coderd.serviceType: LoadBalancer` or `coderd.serviceType: ClusterIP`

Reply:
I'd say that's reasonable.

Reply:
I put these all under a `serviceSpec` variable that just gets injected. Kinda a catch-all.